src/Microsoft.IdentityModel.Tokens/Utility.cs

```diff
@@ -172,37 +170,29 @@ public static bool AreEqual(byte[] a, byte[] b)
         /// <param name="b">
         /// The other set of bytes to compare with.
         /// </param>
-        /// <param name="length">length of array to check</param>
+        /// <param name="length">length of spans to check</param>
         /// <returns>
         /// true if the bytes are equal, false otherwise.
         /// </returns>
-        [MethodImpl(MethodImplOptions.NoOptimization | MethodImplOptions.NoInlining)]
-        internal static bool AreEqual(byte[] a, byte[] b, int length)
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        internal static bool AreEqual(ReadOnlySpan<byte> a, ReadOnlySpan<byte> b, int length)
         {
-            byte[] s_bytesA = new byte[] { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31 };
-            byte[] s_bytesB = new byte[] { 31, 30, 29, 28, 27, 26, 25, 24, 23, 22, 21, 20, 19, 18, 17, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0 };
-
-            int result = 0;
-            int lenToUse = 0;
-            byte[] a1, a2;
-
-            if (((a == null) || (b == null))
-            || (a.Length < length || b.Length < length))
+            if ((a.Length < length || b.Length < length))
```

Comment:
or lengths are unequal?

Reply:
Sorry, I don't understand the question.  I've not changed the logic here... are you asking that it be changed?

Reply:
oh my bad, I misread this.